wlauto/workloads/googleplaybooks/__init__.py

```diff
@@ -0,0 +1,105 @@
+     8. Selects a specified chapter based on page number from the navigation view
+     9. Selects a word in the centre of screen and adds a test note to the page
+    10. Removes the test note from the page (clean up)
+    11. Searches for the number of occurrences of a common word throughout the book
+    12. Switches page styles from 'Day' to 'Night' to 'Sepia' and back to 'Day'
+    13. Uses the 'About this book' facility on the currently selected book
+
+    NOTE: This workload requires a network connection (ideally, wifi) to run.
+    """
+
+    parameters = [
+        Parameter('search_book_title', kind=str, mandatory=False, default="Hamlet",
+                  description="""
+                  The book title to search for within Google Play Books archive.
+                  The book must either be already in the account's library, or free to purchase.
+                  Note: spaces must be replaced with underscores in the book title.
+                  """),
+        Parameter('select_chapter_page_number', kind=int, mandatory=False, default=22,
+                  description="""
+                  The Page Number to search for within a selected book's Chapter list.
+                  Note: Accepts integers only.
+                  """),
+        Parameter('search_word', kind=str, mandatory=False, default='the',
+                  description="""
+                  The word to search for within a selected book.
+                  Note: Accepts single words only.
+                  """),
+        Parameter('markers_enabled', kind=bool, default=True,
+                  description="""
+                  If ``True``, UX_PERF action markers will be emitted to logcat during
+                  the test run.
+                  """),
+    ]
+
+    def validate(self):
+        super(Googleplaybooks, self).validate()
+        self.uiauto_params['package'] = self.package
+        self.uiauto_params['markers_enabled'] = self.markers_enabled
+        self.uiauto_params['book_title'] = self.search_book_title
+        self.uiauto_params['chapter_page_number'] = str(self.select_chapter_page_number)
+        self.uiauto_params['search_word'] = self.search_word
+
+    def initialize(self, context):
+        super(Googleplaybooks, self).initialize(context)
+
+        # This workload relies on the internet so check that there is a working internet connection
+        if not self.device.is_network_connected():
+            raise DeviceError('Network is not connected for device {}'.format(self.device.name))
+
+    def update_result(self, context):
+        super(Googleplaybooks, self).update_result(context)
+
+        # Add classifiers for workload specific input parameters
+        d = {}
+        for param in self.parameters:
+            if param.name not in getattr(AndroidUiAutoBenchmark, "parameters"):
+                d[param.name] = getattr(self, param.name)
+
+        context.add_classifiers(**d)
```

Comment:
Considering that we expect the APK version to be fixed I will remove the update_result method completely and remove the classifiers.